### DISCUSSION TITLE
AWS Lambda configuration

### DISCUSSION THREAD

**Discussion by royassis** [aws/aws-toolkit-jetbrains#3615] [Feature requests]
Hey there

I'm using local **AWS Lambda configuration ** with the SAM cli to build and deploy serverless applications.
Most of my lambda functions are image based (and not zip based), so every lambda has a Dockerfile associated with it.

In the last week wanted to use private built python packages that our team is maintaining, we keep the build artifacts with **CodeArtifact**. This caused a major issue. 

In order to pull and push artifacts into CodeArtifact one needs to use a temporary token issued by CodeArtifact . 
In order for my build to succeed, I need to pass the token every time I run the Pycharm Lambda configuration. 

As I said this token is temporary, and I need to set pass it every time I do an AWS build. That I had to set the **Build Args** parameter in all my Pycharm Lambda configuration configurations, for all my lambda functions, to:
```shell
--parameter-overrides CodeartifactAuthToken=<mytoken>
```
I need to do this every time I refresh the token, which has a max lifetime of 12h.

It would really help if I could set the **Build Args** parameter dynamically, or also set a global **Build Args** parameters for all configurations.

Thanks 
Roy

**Comment by mrichards01** [comment]
@royassis 

Just perusing some of the discussions on here. There is another plugin that may help you in meantime if you're using intellij as your ide of choice. I wondered if you have tried using this plugin - https://plugins.jetbrains.com/plugin/7861-envfile 

Thanks,
Mark

**Reply by royassis** [reply]
Hey @mrichards01, thanks, that looks promising. I'll try to see if It can integrate with the aws lambda configuration.

**Reply by mrichards01** [reply]
I mentioned it because I had a similar issue at my current company which for various reasons required a token to be passed into the local run configurations. It helped to have a script that would run before the run configuration (which I believe you can also enforce in intellij) to pipe a token to an .env file. You could then use that .env file via this plugin to run our application with the necessary token. Perhaps not the answer you are after but should work.
